Source/Core/DiscIO/Volume.cpp

```diff
@@ -73,11 +35,8 @@ std::map<Language, std::string> Volume::ReadWiiNames(const std::vector<u8>& data
     size_t name_end = name_start + NAME_BYTES_LENGTH;
     if (data.size() >= name_end)
     {
-      u16* temp = (u16*)(data.data() + name_start);
-      std::wstring out_temp(NAME_STRING_LENGTH, '\0');
-      std::transform(temp, temp + out_temp.size(), out_temp.begin(), (u16(&)(u16))Common::swap16);
-      out_temp.erase(std::find(out_temp.begin(), out_temp.end(), 0x00), out_temp.end());
-      std::string name = UTF16ToUTF8(out_temp);
+      std::string name = UTF16BEToUTF8(reinterpret_cast<const char16_t*>(data.data() + name_start),
```

Comment:
This dereferencing the pointer result of this cast is undefined behavior by the way (despite being pre-existing).